src/Microsoft.AspNetCore.Mvc.Modules/ModularApplicationPart.cs

```diff
@@ -74,18 +75,23 @@ public IEnumerable<TypeInfo> Types
         {
             get
             {
-                return DefaultAssemblyDiscoveryProvider
-                    .DiscoverAssemblies(
-                        ShellBlueprint.Dependencies.Select(dep => dep.Type.GetTypeInfo().Assembly).Distinct().ToList(),
-                        ReferenceAssemblies)
+                var assemblies = ModularAssemblyProvider
+                    .GetAssemblies(ExtensionLibraryService.RuntimeLibraries);
+
+                var types = DefaultModularAssemblyDiscoveryProvider
+                    .GetCandidateAssemblies(assemblies, ReferenceAssemblies)
                     .SelectMany(x => x.DefinedTypes);
+
+                return types;
             }
         }
 
         /// <inheritdoc />
         public IEnumerable<string> GetReferencePaths()
         {
-            return AssemblyProvider.GetAssemblies().Select(x => x.Location);
+            return DependencyContext.Default.CompileLibraries
```

Comment:
Just answered above, here a copy.

>But for compilation purpose it needs to have references on the netcoreapp (e.g System.Linq). If i remove this line, it doesn't work. I also tried to only add those related to netstandard lybrary but it doesn't fully work.

>But i agree, we could filter specific host compile libraries which are not contained in netcoreapp. But because there are not so many references paths, what you think? Is it worth to do it?